application/src/main/java/org/togetherjava/tjbot/features/cakeday/CakeDayService.java

```diff
@@ -0,0 +1,311 @@
+     *
+     * @param cakeDayRole the Cake Day role to refresh
+     * @param guild the guild in which to refresh Cake Day roles
+     */
+    private void refreshMembersCakeDayRoles(Role cakeDayRole, Guild guild) {
+        guild.findMembersWithRoles(cakeDayRole).onSuccess(members -> {
+            removeRoleFromMembers(guild, cakeDayRole, members);
+            addTodayMembersCakeDayRole(guild);
+        });
+    }
+
+    /**
+     * Asynchronously adds the specified cake day role to guild members who are celebrating their
+     * cake day today.
+     * <p>
+     * The cake day role is added to members who have been in the guild for at least one year.
+     *
+     * @param guild the guild in which to add the cake day role to members
+     */
+    private void addTodayMembersCakeDayRole(Guild guild) {
+        findCakeDaysTodayFromDatabase().forEach(cakeDayRecord -> {
+            UserSnowflake userSnowflake = UserSnowflake.fromId(cakeDayRecord.getUserId());
+
+            int anniversary = OffsetDateTime.now().getYear() - cakeDayRecord.getJoinedYear();
+            if (anniversary > 0) {
+                addCakeDayRole(userSnowflake, guild);
+            }
+        });
+    }
+
+
+    /**
+     * Adds the cake day role to the specified user in the given guild, if available.
+     *
+     * @param snowflake the snowflake ID of the user to whom the cake day role will be added
+     * @param guild the guild in which the cake day role will be added to the user
+     */
+    protected void addCakeDayRole(UserSnowflake snowflake, Guild guild) {
+        Role cakeDayRole = getCakeDayRole(guild).orElse(null);
+
+        if (cakeDayRole == null) {
+            return;
+        }
+
+        guild.addRoleToMember(snowflake, cakeDayRole).complete();
+    }
+
+    /**
+     * Adds the cake day role to the specified member if the cake day role exists in the guild.
+     *
+     * @param member the {@link Member} to whom the cake day role will be added
+     */
+    protected void addCakeDayRole(Member member) {
+        Guild guild = member.getGuild();
+        UserSnowflake snowflake = UserSnowflake.fromId(member.getId());
+        Role cakeDayRole = getCakeDayRole(guild).orElse(null);
+
+        if (cakeDayRole == null) {
+            return;
+        }
+
+        guild.addRoleToMember(snowflake, cakeDayRole).complete();
+    }
+
+    /**
+     * Removes a specified role from a list of members in a guild.
+     *
+     * @param guild the guild from which to remove the role from members
+     * @param role the role to be removed from the members
+     * @param members the list of members from which the role will be removed
+     */
+    private void removeRoleFromMembers(Guild guild, Role role, List<Member> members) {
+        members.forEach(member -> {
+            UserSnowflake snowflake = UserSnowflake.fromId(member.getIdLong());
+            guild.removeRoleFromMember(snowflake, role).complete();
+        });
```

Comment:
Can't you chain these RestActions?

Reply:
`RestActions` are now handled better! :)